common/lib/xmodule/xmodule/studio_editable.py

```diff
@@ -55,3 +55,45 @@ class StudioEditableDescriptor(object):
     """
     author_view = module_attr(AUTHOR_VIEW)
     has_author_view = True
+
+
+class StudioEditableBlock(object):
+    """
+    Helper methods for supporting Studio editing of XBlocks.
+
+    This class is only intended to be used with an XBlock!
+    """
+    has_author_view = True
+
+    def render_children(self, context, fragment, can_reorder=False, can_add=False):
+        """
+        Renders the children of the module with HTML appropriate for Studio. If can_reorder is True,
+        then the children will be rendered to support drag and drop.
+        """
+        contents = []
+
+        for child in self.get_children():  # pylint: disable=no-member
+            if can_reorder:
+                context['reorderable_items'].add(child.location)
+            child_module = self.system.get_module(child)  # pylint: disable=no-member
+            rendered_child = child_module.render(StudioEditableModule.get_preview_view_name(child_module), context)
+            fragment.add_frag_resources(rendered_child)
+
+            contents.append({
+                'id': child.location.to_deprecated_string(),
+                'content': rendered_child.content
+            })
+
+        fragment.add_content(self.system.render_template("studio_render_children_view.html", {  # pylint: disable=no-member
+            'items': contents,
+            'xblock_context': context,
+            'can_add': can_add,
+            'can_reorder': can_reorder,
+        }))
+
+    @staticmethod
+    def get_preview_view_name(block):
+        """
+        Helper method for getting preview view name (student_view or author_view) for a given module.
+        """
+        return AUTHOR_VIEW if hasattr(block, AUTHOR_VIEW) else STUDENT_VIEW
```

Comment:
Why duplicate all this code, which is in StudioEditableModule except for `has_author_view`? Why not a common mixin?